src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -163,8 +166,13 @@ public int numberOfShards() {
     }
 
     @Override
-    public UnmodifiableIterator<IndexShard> iterator() {
-        return shards.values().iterator();
+    public Iterator<IndexShard> iterator() {
```

Comment:
I don't understand, why would we want to allow someone to remove a shard via `.remove()` on the iterator?